### ISSUE TITLE
TR feedback

### ISSUE THREAD

**Issue by matyaskopp** [clarin-eric/ParlaMint#508]
## Taxonomies are missing translations
- [x] taxonomies translation

This will be required in v3.1, but you can do it now

## rename NER taxonomy
- [x] NER taxonomy filename + id

`ParlaMint-taxonomy-NER` -> `ParlaMint-taxonomy-NER.ana`

## meeting should correspond to data
- [x] TEI root file `meeting` content
- [x] TEI.ana root file `meeting` content

I like you have encoded the full history of parliament organization, but the `meeting` element should contain just terms that are included in text data:
https://github.com/coltekin/ParlaMint/blob/aa80f96ccaa80f0dd4de0bf250be37284c6f3d29/Data/ParlaMint-TR/ParlaMint-TR.xml#L10-L36
should be
```XML
            <title type="sub" xml:lang="en">Minutes of the Grand National Assembly of Turkey, terms 24 - 27 (2011 - 2022)</title>
            <title type="sub" xml:lang="tr">Türkiye Büyük Millet Meclisi tutanakları, 24 - 27. dönemler (2011 - 2022)</title>
            <meeting n="27" corresp="#TBMM" ana="#parla.term #TBMM.27"/>
            <meeting n="26" corresp="#TBMM" ana="#parla.term #TBMM.26"/>
            <meeting n="25" corresp="#TBMM" ana="#parla.term #TBMM.25"/>
            <meeting n="24" corresp="#TBMM" ana="#parla.term #TBMM.24"/>
```

## bibl timespan does not correspond to `title` and `setting` timespan
- [x] corpus timespan
https://github.com/coltekin/ParlaMint/blob/aa80f96ccaa80f0dd4de0bf250be37284c6f3d29/Data/ParlaMint-TR/ParlaMint-TR.ana.xml#L72

```XML
               <date from="2009-04-01" to="2022-10-20"/>
```
vs:
```XML
<title type="sub" xml:lang="en">Minutes of the Grand National Assembly of Turkey, terms 24 - 27 (2011 - 2022)</title>
```
```XML
            <setting>
               <name type="city">Ankara</name>
               <name type="country" key="TR">Turkey</name>
               <date from="2011-10-13" to="2022-02-24"/>
            </setting>
```

## space in app version
- [x] `application/@version`

https://github.com/coltekin/ParlaMint/blob/aa80f96ccaa80f0dd4de0bf250be37284c6f3d29/Data/ParlaMint-TR/ParlaMint-TR.ana.xml#L143
```XML
            <application ident="steps-parser" version=" 7c79014">
```

## TEI version includes NER taxonomy
- [x] NER taxonomy in TEI version

(this caused the wrong filename when you factorized `teiHeader`)
https://github.com/coltekin/ParlaMint/blob/aa80f96ccaa80f0dd4de0bf250be37284c6f3d29/Data/ParlaMint-TR/ParlaMint-TR.xml#L118-L119
```XML
            <xi:include xmlns:xi="http://www.w3.org/2001/XInclude"
                        href="ParlaMint-taxonomy-NER.xml"/>
```

## bibl title
- [x] bill `title` and `idno`

https://github.com/coltekin/ParlaMint/blob/aa80f96ccaa80f0dd4de0bf250be37284c6f3d29/Data/ParlaMint-TR/2011/ParlaMint-TR_2011-10-13-tbmm-T24.xml#L38-L43
the content of title:
```XML
          <title type="main" xml:lang="en">https://www5.tbmm.gov.tr/tutanak/donem24/yil2/bas/b007m.htm</title>
```
should be moved to `idno`
```XML
<idno type="URI" subtype="parliament">https://www5.tbmm.gov.tr/tutanak/donem24/yil2/bas/b007m.htm</idno>
```
## one component file content - session?
- [x] `session`/`meeting`/`sitting`
- [x] unique title

I am not sure what one file contains:
```XML
<?xml version='1.0' encoding='utf-8'?>
<TEI xmlns="http://www.tei-c.org/ns/1.0" xml:id="ParlaMint-TR_2011-10-13-tbmm-T24" xml:lang="tr" ana="#parla.session #reference">
  <teiHeader>
    <fileDesc>
      <titleStmt>
        <title type="main" xml:lang="en">The Turkish parliamentary corpus ParlaMint-TR, Term 24, Session 7 [ParlaMint SAMPLE]</title>
        <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 24, Oturum 7 [ParlaMint SAMPLE]</title>
        <meeting n="24" corresp="#TBMM" ana="#parla.term #TBMM.24">Term 24</meeting>
        <meeting n="7" ana="#parla.session">7</meeting>
```
 but I guess it is a sitting, not a session, so I expect to encode it this way:

```XML
<?xml version='1.0' encoding='utf-8'?>
<TEI xmlns="http://www.tei-c.org/ns/1.0" xml:id="ParlaMint-TR_2011-10-13-tbmm-T24" xml:lang="tr" ana="#parla.sitting #reference">
  <teiHeader>
    <fileDesc>
      <titleStmt>
        <title type="main" xml:lang="en">The Turkish parliamentary corpus ParlaMint-TR, Term 24, Meeting 7, 2011-10-13  [ParlaMint SAMPLE]</title>
        <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 24, Oturum 7, 2011-10-13 [ParlaMint SAMPLE]</title>
        <meeting n="24" corresp="#TBMM" ana="#parla.term #TBMM.24">Term 24</meeting>
        <meeting n="7" ana="#parla.meeting">7</meeting>
        <meeting n="2011-10-13" ana="#parla.sitting">2011-10-13</meeting>
```

furthermore main title is not unique:
```bash
find . -type f -name "ParlaMint-TR_*"|xargs cat|grep '<title type="main" xml:lang="tr">'|sort|uniq -c|sort -nr|head
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 9 [ParlaMint]</title>
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 8 [ParlaMint]</title>
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 7 [ParlaMint]</title>
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 6 [ParlaMint]</title>
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 5 [ParlaMint]</title>
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 4 [ParlaMint]</title>
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 3 [ParlaMint]</title>
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 2 [ParlaMint]</title>
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 10 [ParlaMint]</title>
      6         <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Oturum 1 [ParlaMint]</title>
```

**Comment by matyaskopp** [comment]
> Thank you for the feedback! I am almost all done, but before pushing the new sample, I have a question: is the taxonomy/hierarchy of the meetings supposed to be uniform over all ParlaMint corpora?

Yes, I am trying to push everyone to stick and use the taxonomy (`term`, `session`, `meeting` and `sitting`). In some countries does not make some classes sense (in CZ, we don't have `session`s  so I am not using them)

> * `parla.meeting`: one day of debates (this is the main division used in the transcripts)
> * `parla.sitting`: debates without a break (one "meeting" may consist of multiple "sittings", these are also always marked in the transcripts)

How I see it:
`parla.meeting` is defined as a list of topics that should be discussed. The topic list is a plan of what MPs want to discuss, but usually, they are not able to discuss everything in one (`parla.sitting`)day because of various reasons:
- not enough time
- an important person for a discussed topic is missing

`parla.sitting` you are generally right, but I have to add that "small breaks" are allowed (parliamentary groups can ask for a short break if they want to consult together)

I think if a meeting consists of one sitting, then the file should be marked both with meeting and sitting. 
And we are dividing files by date (so sitting is the proper level for one file).

I hope @TomazErjavec sees it in the same way.

**Comment by coltekin** [comment]
Thank you again for the quick responses.

Normally, I only need to refer to "term", which is clear and unambiguous, and the gatherings that covers a full-day. The daily gatherings (tr: "birleşim") has an official definition, and assigned numbers. Transcripts are divided by these units, and I also take them as file boundaries. There are subdivisions marked within each day-meeting ("oturum", literally: sittings), but I do not need to refer to them in the corpus. The motivation for counting every legislative year as a "session" comes from the fact that, daily meeting numbers reset at every legislative year (the name also matches Wikipedia's definition of [Legislative session](https://en.wikipedia.org/wiki/Legislative_session)). Without the session, the numbering in the `titleStmt` becomes ambiguous. So, I'd like to have something like:
```
      <titleStmt>
        <title type="main" xml:lang="en">The Turkish parliamentary corpus ParlaMint-TR, Term 27, Year 5, Meeting 58 [ParlaMint SAMPLE]</title>
        <title type="main" xml:lang="tr">Türkiye meclis derlemi ParlaMint-TR, Dönem 27, Yıl 5, Birleşim 58 [ParlaMint SAMPLE]</title>
        <meeting n="27" corresp="#TBMM" ana="#parla.term #TBMM.27">Term 27</meeting>
        <meeting n="5" ana="#parla.session">58</meeting>
        <meeting n="58" ana="#parla.meeting">58</meeting>
```

However, I am not attached to this particular hierarchy, and I'd be happy to implement if there is a more uniform approach covering this sort of division.

**Comment by TomazErjavec** [comment]
Thanks for the explanation, it makes perfect sense & I stand corrected, so pls. go ahead. You could use divs inside a file, if you wanted to, cf. [Guidelines](https://clarin-eric.github.io/ParlaMint/#chp-div), although I have to admit the overall structure of a corpus could be much better explained, as much as we understand it...

I can't seem to point to TR data, regardless of the branch?
This should be a separate issue, to TR (or any other) data that is not yet perfect, @matyaskopp. 
I'd suggest that what passes validation goes to main, and *everything* submitted is merged into data, https://github.com/clarin-eric/ParlaMint/tree/data/Data/ParlaMint-TR, in this case. And https://github.com/clarin-eric/ParlaMint/tree/data/Data/ParlaMint-TR is months old, can't these branches be delete, if stale? 
In short too many and to conservative branches, git should be more flexi I think. 
Sorry, a night rave, will go watch movie now :)

**Comment by coltekin** [comment]
* [ ]  taxonomies translation: partially done, except UD taxonomy (linguistic terminology) which is rather tricky. I will complete the translations for v3.1.
* [X]  NER taxonomy filename + id
* [X]  TEI root file `meeting` content
* [X]  TEI.ana root file `meeting` content
* [X]  corpus timespan
* [X]  `application/@version`
* [X]  NER taxonomy in TEI version
* [X]  bill `title` and `idno`
* [X]  `session`/`meeting`/`sitting`: each component is now a `meeting`
* [X]  unique title: this was related to above problem. Meeting number alone is ambiguous, adding year (session) disambiguates.

I have just pushed the changes to my fork, the PR should now be updated (#499). The sample can be viewed through my fork at <https://github.com/coltekin/ParlaMint/tree/data/Data/ParlaMint-TR> before merge.

This version of the pipeline should also fix UD validation issues I received via email (duplicate IDs and self loops in conllu). In case these errors occur in other corpora: the issue in my case was caused by a bug in my pipeline that duplicated some sentences (using the same sentence ID). In my local validation and conllu generation, the errors did not show up. However, it seems they are shown in case I try to parallelize make (by -j).

I will run the conversion and validation on the complete corpus tonight.

**Comment by coltekin** [comment]
The latest version of the full corpus is now at <http://sfs.uni-tuebingen.de/~ccoltekin/parlamint/>. I run the validation and conllu generation locally without any errors, checked the issues above. I'd be happy to fix if there are any remaining errors.

**Comment by matyaskopp** [comment]
> [ ] taxonomies translation: partially done, except UD taxonomy (linguistic terminology) which is rather tricky. I will complete the translations for v3.1.

@coltekin 
**ud-syn taxonomy translation is not required !!!** We are planning to use one common UD-SYN taxonomy for all corpora:
https://github.com/clarin-eric/ParlaMint/blob/main/Data/Taxonomies/ParlaMint-taxonomy-UD-SYN.ana.xml
And it would be a real nightmare to translate it into all languages.
So it will be the only taxonomy without translation.

**Comment by coltekin** [comment]
Happy to hear this :-). The translations for other taxonomies should already be done. If I overlooked any, I will complete  them for v3.1.

**Comment by TomazErjavec** [comment]
>> The latest version of the full corpus is now at http://sfs.uni-tuebingen.de/~ccoltekin/parlamint/.

> Thanks,will convert & let you know.

@coltekin, the latest log is, as before, at https://nl.ijs.si/et/tmp/ParlaMint/Repo/ParlaMint-TR.log
As far as I see (grep -i warn, and then sucessive grep -v), the warnings left are:
*  replacing the number of words 
* fixing the covid subcorpus (should be=>1.11.2019)
* political parties without anybody affiliated to them, e.g. `politicalParty-role organisation without affiliation: #party.MSP`
* some time-related info, e.g. `WARN[08] ParlaMint-TR-listPerson Affiliate from date (2011-06-28T00:00:00) is P475D before party.HDP organization beginning (2012-10-15T00:00:00)`

I think none of these is a deal breaker for 3.0.

**Comment by coltekin** [comment]
> @coltekin, the latest log is, as before, at https://nl.ijs.si/et/tmp/ParlaMint/Repo/ParlaMint-TR.log As far as I see (grep -i warn, and then sucessive grep -v), the warnings left are:
> 
>     * replacing the number of words

My counting was also a bit off for the earlier versions. I'd be happy to check this later, but since your release process unifies them across all ParlaMint corpora, I did not pay a lot of attention.

>     * fixing the covid subcorpus (should be=>1.11.2019)

I fixed this in my template. It will be correct when I re-run the conversion.

>     * political parties without anybody affiliated to them, e.g. `politicalParty-role organisation without affiliation: #party.MSP`

These are here because I am experimenting with including some older data (extracted from PDF files). I can make sure that only parties with affiliations are listed in the next version of the corpus.

>     * some time-related info, e.g. `WARN[08] ParlaMint-TR-listPerson Affiliate from date (2011-06-28T00:00:00) is P475D before party.HDP organization beginning (2012-10-15T00:00:00)`

This is the trickiest. I omitted the `from` for party affiliations if I cannot determine it. As a result, the validation warns for parties that are founded within the period covered by the corpus. Setting `from` to the foundation date of the party would fix the warning, but it may not reflect the reality.

I think this was also the consensus out of the discussion in #397.

> I think none of these is a deal breaker for 3.0.

Then, unless other issues/bugs discovered before the release, we can use this version for 3.0. If I improve or fix anything important, I will upload the data to the same location and let you know.

**Comment by TomazErjavec** [comment]
>> * political parties without anybody affiliated to them, e.g. `politicalParty-role organisation without affiliation: #party.MSP`

> These are here because I am experimenting with including some older data (extracted from PDF files). I can make sure that only parties with affiliations are listed in the next version of the corpus.

I think it is ok if you leave them, they do not hurt, so why get rid of information. Maybe at some point you want to extend the corpus into the past...

As for the rest, I agree with @matyaskopp , it is ok to consider TR 3.0 ready, except, if you would have a look at the data under the concordancer if you spot any problems. The entry point is on our development concordancer: https://www.clarin.si/noske-beta/parlamint30.cgi/ (in particular the [info page](https://www.clarin.si/noske-beta/parlamint30.cgi/corp_info?corpname=parlamint30_tr&struct_attr_stats=1&subcorpora=1)).
Username is dev, password alfabetagama (no great secret, just so people don't wind up here instead of the production one by mistake). 
Just let us know, and then we can close this issue.
